nixos/lib/test-driver/test_driver/logger.py

```diff
@@ -77,6 +78,15 @@ def drain_log_queue(self) -> None:
         except Empty:
             pass
 
+    @contextmanager
+    def paused(self) -> Iterator[None]:
+        """
+        Suspends all logging operations
```

Comment:
"paused" seems a bit misleading, as i understand that the logger is switched into a mode where it *drops* all messages?

Reply:
It seems like this change is purely for cosmetical reasons, is it?

Reply:
Exactly, see the commit [message](https://github.com/NixOS/nixpkgs/pull/214910/commits/3fbba6ff51cfcc362cbcd4512fbf5f51f5c62cfd).

Reply:
Hm if it's just for cosmetics, I'd just drop this specific patch.
The problem that i see is that we have an accumulating set of patches that do little adaptions here and there but in their sum they undermine architectural goals.

The right way to do this would be introducing a real logging framework that handles verbosity levels. Until then, I think we can live with logs that are a bit too verbose. What do you think?

Reply:
> The right way to do this would be introducing a real logging framework that handles verbosity levels

Agreed, although in this specific case I wouldn't log at all. `send_chars` using `send_key` is totally an implementation detail that the developer of a test doesn't need to know at any logging level.

Reply:
Then let's simply drop the log message of the send_key function and if someone needs it for debugging, then they would add a log call there again.

Reply:
Uhm, I can't just do that: the source of the log lines are actually `send_monitor_command` and `wait_for_monitor_prompt`. Should I remove the `self.nested` call from them?
I think it makes sense because they're very low-level functions and probably shouldn't even be public.

Reply:
Yes, let's drop it. This is a python script which is very easy to extend with log lines in case someone needs to debug something. It's easy to reintroduce logging, so let's not build strange constructs that make this selectively with unclear goals.